Source/ACE/Entity/Player.cs

```diff
@@ -2343,6 +2355,76 @@ public void HandleActionStackableSplitToContainer(uint stackId, uint containerId
         }
 
         /// <summary>
+        /// This method is used to remove X number of items from a stack, including 
+        /// If amount to remove is greater or equal to the current stacksize, item will be removed.
+        /// </summary>
+        /// <param name="stackId">Guid.Full of the stacked item</param>
+        /// <param name="containerId">Guid.Full of the container that contains the item</param>
+        /// <param name="amount">Amount taken out of the stack</param>
+        /// <returns></returns>
+        public void HandleActionRemoveItemFromInventory(uint stackId, uint containerId, ushort amount)
+        {
+            ActionChain removeItemsChain = new ActionChain();
+            removeItemsChain.AddAction(this, () =>
+            {
+                Container container;
+                if (containerId == Guid.Full)
+                    container = this;
+                else
+                    container = (Container)GetInventoryItem(new ObjectGuid(containerId));
+
+                if (container == null)
+                {
+                    log.InfoFormat("Asked to remove an item {0} in container {1} - the container was not found",
+                        stackId,
+                        containerId);
+                    return;
+                }
+
+                WorldObject item = container.GetInventoryItem(new ObjectGuid(stackId));
+                if (item == null)
+                {
+                    log.InfoFormat("Asked to remove an item {0} in container {1} - the item was not found",
+                        stackId,
+                        containerId);
+                    return;
+                }
+
+                if (amount >= item.StackSize)
+                    amount = (ushort)item.StackSize;
+
+                ushort oldStackSize = (ushort)item.StackSize;
+                ushort newStackSize = (ushort)(oldStackSize - amount);
+
+                if (newStackSize < 1)
+                {
+                    // Remove item from inventory
```

Comment:
add this before the other two lines
`Session.Network.EnqueueSend(new GameMessageSetStackSize(item.Sequences, item.Guid, 0, 0));
`

Reply:
I think it will fix your remove bug